src/django/VLE/models/journal_content/comment.py

```diff
@@ -10,79 +10,142 @@
 
-        If include_others, include comments that are not attached to the feedback node
-        """
-        if not feedback_form.can_view_published_feedback_comments(user):
-            return self.none()
+        If the feedback_form is set, viewable comments related to the feedback form will also be shown"""
+        journal, feedback_form = context
+        is_grader = user.has_permission('can_grade', journal.assignment)
 
-        if feedback_form.preset_peer_feedback_node:
-            include_others = include_others and feedback_form.preset_peer_feedback_node.can_view_other_comments
+        # Graders should be able to see all comments, also unpublished reviewer comments, and comments
+        # that are not attached to the feedback form
+        if is_grader:
+            return self
 
-        entries_related_to_fbf_qry = VLE.models.Entry.objects.filter_on_feedback_form(feedback_form)
+        is_in_journal = journal.is_in_journal(user)
+        if not is_in_journal and not is_grader and not feedback_form:
+            raise VLEForbiddenError('You are required to provide a feedback form to view comments of this journal.')
 
-        q = self.build_q_through_node(
-            entry__in=entries_related_to_fbf_qry,
-            available_node_classes=VLE.models.Entry.get_available_node_classes(),
-        )
-        self = self.filter(
-            q
-            | Q(preset_peer_feedback_node=feedback_form.preset_peer_feedback_node)
-            | Q(preset_entry_node__in=VLE.models.PresetEntryNode.objects.filter_on_feedback_form(
-                feedback_form=feedback_form,
-                entry_qry=entries_related_to_fbf_qry,
-            ))
-        )
+        self = self.filter_based_on_published_state(feedback_form, is_in_journal=is_in_journal, is_grader=is_grader)
+        if feedback_form:
+            self = self.filter_based_on_feedback_form(feedback_form)
 
-        if include_others:
-            self = self.filter(
-                Q(feedback_form=feedback_form) | Q(published=True)
-            )
+        return self.distinct('pk')
+
+    def filter_based_on_published_state(
+        self,
+        feedback_form: Union['VLE.models.FeedbackForm', None],
+        is_in_journal: bool,
+        is_grader: bool,
+    ):
+        is_published = self.build_q_is_published()
+        # Journal authors should only be able to view published comments
+        if is_in_journal:
+            return self.filter(is_published)
+        # Graders should be able to see all comments
+        elif is_grader:
+            return self
+        # Reviewers should see:
+        # - all comments within the provided feedback form
+        # - all other published comments
         else:
-            self = self.filter(
-                feedback_form=feedback_form,
+            if not feedback_form:
+                raise VLEForbiddenError('You are required to provide a feedback form to view comments of this journal.')
```

Comment:
[NIT] As this explicitly regards reviewers, we could use that in the message. 

```suggestion
                raise VLEForbiddenError('You need to be invited to review this journal, to view any of its comments.')
```